plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection/ProductCollectionController.php

```diff
@@ -737,6 +358,7 @@ public function add_support_for_filter_blocks( $pre_render, $parsed_block ) {
 		}
 
 		$this->parsed_block = $parsed_block;
+		$this->renderer->set_parsed_block( $parsed_block );
```

Comment:
We use `parsed_block` in both this class and `ProductRenderer`, which is why we utilize the setter here. While this might not be the ideal solution, I aimed to maintain simplicity by primarily relocating code and making minimal modifications in this PR.